bitcoin/examples/taproot-psbt.rs

```diff
@@ -49,7 +49,7 @@ const UTXO_1: P2trUtxo = P2trUtxo {
     script_pubkey: UTXO_SCRIPT_PUBKEY,
     pubkey: UTXO_PUBKEY,
     master_fingerprint: UTXO_MASTER_FINGERPRINT,
-    amount_in_sats: 50 * COIN_VALUE, // 50 BTC
+    amount_in_sats: Amount::from_sat(50 * 100_000_000), // 50 BTC
```

Comment:
Recently @Kixunil made a push for using a custom integer format for bitcoin values `1_000_000_00`, requires 
```
    #[allow(clippy::inconsistent_digit_grouping)] // Group to show 100,000,000 sats per bitcoin.
```
You can see a bunch of these in `amount.rs`. You could use it if you like it.

Reply:
I guess we should have `Amount::from_whole_btc` accepting `u64`. But also `50 * Amount::ONE_BTC` should work.

Reply:
I tried both `50 * Amount::ONE_BTC` and `Amount::from_str("50 BTC").unwrap()` here but it's not currently possible because this is a `const` so the compiler doesn't allow it.

Reply:
We should probably have `#![allow(clippy::inconsistent_digit_grouping)]` on the whole crate. This lint is stupid and has a 100% false positive rate for me.

Reply:
Why are 10, 25, and 100 interesting? Also for powers of 10 sats, maybe just the commonly used units? `MILLI_BTC`, `MEGA_SAT`... There should be little reason for arbitrary integers if you can simply call `from_sat`. 0, 1 sat, 1 btc, 21M btc are "magic numbers" so those do make sense.

Reply:
Regarding the lint, ideally we would have it only off for numbers passed into `Amount::from_sat` but that will probably never happen.

Reply:
@Kixunil they're interesting because we have to do dumb workarounds in const code otherwise (e.g. multiplying by the number 100_000_000 and then commenting the units), and because they have nearly no cost to us. In particular 50 BTC is used all over the codebase.

10 and 100 are powers of ten, which makes it useful when constructing units in human terms. Maybe we should have 1000 and 10000 too, though the usefulness degrades as we get away from "sane" numbers of BTC.

50 is the default coinbase amount on regtest networks and in the first 4 years of Bitcoin's life.

25 I will grant you has no particular use.

Reply:
We can have non-trait `const` methods (like `from_whole_btc` I suggested).

Yeah, I understand 50, so I wouldn't mind something like `BASE_SUBSIDY` (not sure about the name).

Reply:
The problem with a from_btc const method is unwrap is "not yet stable as a const fn".  For example:

```
pub const fn from_int_btc(btc: u64) -> Amount {
    Amount(btc.checked_mul(100_000_000 * btc).unwrap())
}
```
I guess this would be fine if we didn't check for overflow or manually implemented the overflow check.

Reply:
otherwise @apoelstra idea of just making the most common needed const amount would work.

Reply:
`from_int_btc_unchecked` can be const. The other would be `fn from_int_btc(btc: u64) -> Option<Amount>`

Reply:
@yancyribbens we could return `Option` or use this workaround:

```rust
match btc.checked_mul(100_000_000) {
    Some(amount) => Amount::from_sat(amount),
    None => {
        // We could also panic in newer versions of Rust to make the message nicer.
        #[allow(/* don't remember the name, the compiler will tell you when you try */)]
        let _integer_overflow_converting_btc_to_sats = [(); 0][1];
        Amount(0) // unreachable but we have to return something to satisfy type checker
    },
}
```

Also if we're panicking we should use `#[track_caller]` on the method.

Reply:
I think you mean: `const fn from_int_btc(btc: u64) -> Option<Amount>`?  And if this is returning an Option it can't be unwrapped in another const fn, right?

Reply:
Since lack of `unwrap()` is so annoying I prefer to use the workaround I suggested.

Reply:
yes, but it's turtles all the way down.  Your solution to manually implement `unwrap()` works for this function, but the calling function would also need to manually implement `unwrap()` and so on.  Also, `0` is a bit of a hack in that it doesn't convey their was an error really.  Personally I think maybe we just add `const FIFTY` for now and wait for `unwrap()` to be available in const.

Reply:
ah sorry I see now. Yes that should work, except can't we just use `panic!` with a message?  Panic should be available in the MSRV?

Reply:
@Kixunil in rust-bech32 we are panicking in a constfn; see discussion here https://github.com/rust-bitcoin/rust-bech32/pull/103#discussion_r1187449796

I think in 1.48 it's ok?

Reply:
> But we could conditionally use it to provide a better error message in higher versions.

Do we have examples of version conditionals already in the code base?  It looks like things such as `rustversion` require an added dependency.

Reply:
Added a PR here for this https://github.com/rust-bitcoin/rust-bitcoin/pull/1870